### PR TITLE
Issue #386: Add direct model allowing pass through to `brms`

### LINKED ISSUE #386: Create pass through model
It's a good test of our infrastructure that we can pass through to `brms` with as little additional model code as possible.

Doing this will also allow users to test our models in comparison to a baseline with no adjustments for biases in the data.

Some things I can spot we need to do to achieve this:

* `epidist_prior` passes through cleanly
* `epidist_stancode` passes through cleanly
* `epidist_validate` works for pass through model
* Some kind of shared preprocessing for pass through model

### PR DESCRIPTION
**Pull request by athowes**
## Description

This PR closes #386.

It adds a pass through model called `direct_model` and the infrastructure needed to support it. It also adds tests for that model (both unit and integration).

Remaining things to discuss IMO:

* In the class constructor for `direct_model` I call the times `ptime` and `stime` -- this seems debateable: https://github.com/epinowcast/epidist/issues/402
* Consider name different to `direct_model`: https://github.com/epinowcast/epidist/issues/400
* Follow up issue on using this in the getting started vignette to show why our models (e.g. `latent_individual` are better): https://github.com/epinowcast/epidist/issues/403

## Checklist

- [x] My PR is based on a package issue and I have explicitly linked it.
- [x] I have included the target issue or issues in the PR title in the for Issue(s) *issue-numbers*: PR title
- [x] I have read the [contribution guidelines](https://github.com/epinowcast/.github/blob/main/CONTRIBUTING.md).
- [x] I have tested my changes locally.
- [x] I have added or updated unit tests where necessary.
- [x] I have updated the documentation if required.
- [x] My code follows the established coding standards.
- [ ] I have added a news item linked to this PR.
- [x] I have reviewed CI checks for this PR and addressed them as far as I am able.

### CODE REVIEW

**Comment by seabbs** [comment]
Before I review can you follow up on where you have landed with "Remaining things to discuss IMO" points?

**Comment by athowes** [comment]
> Before I review can you follow up on where you have landed with "Remaining things to discuss IMO" points?

1. In the class constructor for direct_model I call the times ptime and stime -- this seems debateable: I think this is something we can figure out during the proprocessing work. As it sort of relates to e.g. taking `epidist_linelist` as input here

2. Consider name different to direct_model: I'm fine with `direct_model`. One aspect that is dislikeable about it is that it ends in model. Whereas `latent_individual` doesn't. So maybe it should just be `direct`. There are lots of other possible options but I'm interested to hear your POV
 
3. Follow up issue on using this in the getting started vignette to show why our models (e.g. latent_individual are better): yes this is something I can make

File: R/direct_model.R [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+#' @export
+as_direct_model <- function(data) {
+  UseMethod("as_direct_model")
+}
+
+assert_direct_model_input <- function(data) {
+  assert_data_frame(data)
+  assert_names(names(data), must.include = c("case", "ptime", "stime"))
+  assert_integer(data$case, lower = 0)
+  assert_numeric(data$ptime, lower = 0)
```
Comment (athowes) [inline R/direct_model.R:14]: Could be `_lwr`

File: R/direct_model.R [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+#' @rdname as_direct_model
+#' @method as_direct_model data.frame
+#' @family direct_model
+#' @autoglobal
+#' @export
+as_direct_model.data.frame <- function(data) {
+  assert_direct_model_input(data)
+  class(data) <- c("epidist_direct_model", class(data))
+  data <- data |>
+    mutate(delay = .data$stime - .data$ptime)
```
Comment (athowes) [inline R/direct_model.R:37]: Naive version of delay (not lower or upper bound on delay).

**Comment by seabbs** [comment]
Can you give a ping when this is ready for re-review as the tracker has got a bit confused

**Comment by athowes** [comment]
> Can you give a ping when this is ready for re-review as the tracker has got a bit confused

Ready now. Should have things merged in, and added issues for all follow-up.

**Review by seabbs — APPROVED** [review]

**Pull request merged**